src/ucp/rndv/rndv.c

```diff
@@ -177,13 +169,21 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_proto_progress_rndv_rtr, (self),
 
     /* send the RTR. the pack_cb will pack all the necessary fields in the RTR */
     packed_rkey_size = ucp_ep_config(rndv_req->send.ep)->rndv.rkey_size;
-    status = ucp_do_am_single(self, UCP_AM_ID_RNDV_RTR, ucp_rndv_rtr_pack,
-                              sizeof(ucp_rndv_rtr_hdr_t) + packed_rkey_size);
-    if (status == UCS_OK) {
-        /* release rndv request */
+    status           = ucp_do_am_single(self, UCP_AM_ID_RNDV_RTR,
+                                        ucp_rndv_rtr_pack,
+                                        sizeof(ucp_rndv_rtr_hdr_t) +
+                                        packed_rkey_size);
+    if ((status != UCS_OK) && (status != UCS_ERR_NO_RESOURCE)) {
+        ucs_hlist_del(&ucp_ep_ext_gen(rndv_req->send.ep)->proto_reqs,
```

Comment:
you would not need this, if adding to list will be done here on success

Reply:
yep
done
but we still need this for handling UCT/SELF TL ... that could provide us ATP from AM operation context